### PR TITLE
Add test for broken cmap

### PR DESCRIPTION
**Pull request by qwrtln**
This is to get quick feedback if Ghostscript breaks the file's CMap again.

With Ghostscript 10.04 all the files built with LuaTeX are broken, in which case the test should fail.

### CODE REVIEW

File: .github/workflows/build-and-publish-rule-book.yaml [hunk 1]
```diff
@@ -96,6 +96,19 @@ jobs:
           NUM_OF_PAGES=$(pdfinfo main_${{ matrix.language }}.pdf | awk '/^Pages:/ {print $2}')
           echo "NUM_OF_PAGES=${NUM_OF_PAGES}" >> "$GITHUB_OUTPUT"
 
+      - name: Check for broken CMap
+        if: github.event_name == 'pull_request'
+        run: |
+          QUOTE=$(grep -A1 heegusquote main_${{ matrix.language }}.tex | tail -1 | grep -o "{.*}" | tail -c +2 | head -c -2)
```
Comment (DarthGandalf) [inline .github/workflows/build-and-publish-rule-book.yaml:102]: this tail|head deserves a comment

**Review by DarthGandalf — APPROVED** [review]

**Comment by qwrtln** [comment]
What the heck, French? For whatever reason, the test is failing.

File: .github/workflows/build-and-publish-rule-book.yaml (hunk 1 above)
Comment (qwrtln) [reply .github/workflows/build-and-publish-rule-book.yaml:102]: Either that, or maybe you know a better way of getting rid of the {} surrounding the text? Not a fan of this tail|head.

**Comment by qwrtln** [comment]
OK, French is splitting the quote into two lines
```
J’ai réalisé ce projet par amour de l’apprentissage des jeux et de la compréhension de leur
complexité.
```

**Comment by DarthGandalf** [comment]
```
cat main_fr.tex | perl -E 'local $/; $x=<>; $x =~ /heegusquote.*?textit\{(.*?)\}/s; print $1'
```

**Comment by qwrtln** [comment]
> ```
> cat main_fr.tex | perl -E 'local $/; $x=<>; $x =~ /heegusquote.*?textit\{(.*?)\}/s; print $1'
> ```

I had to change the line because French was having a problem again with quotes. They're rendered differently in the PDF file. This time it's without curly braces, so no need for Perl, I hope that's OK.

**Comment by DarthGandalf** [comment]
> no need for Perl, I hope that's OK

It's not like I'm married to perl, or something.

Readability-wise, new version is better than before, but it seems even more fragile. It depends on the specific line which is 2 lines below the "intro" to not have any LaTeX commands in any language, right?

**Comment by qwrtln** [comment]
> Readability-wise, new version is better than before, but it seems even more fragile. It depends on the specific line which is 2 lines below the "intro" to not have any LaTeX commands in any language, right?

Maybe let me state the problem first. Files built with LuaTeX and optimized with Ghostscript look good, but if you try to convert them to txt or copy any text, you get stuff like this:
```
"v h?2 *QKKmMBiv
P+iQ#2` 9- kyk9
o2`bBQM RXy
h?Bb Bb  +QKKmMBiv@/`Bp2M T`QD2+i- r?B+? ?b  :Bi>m# `2TQbBiQ`vX 1p2`vQM2 Bb r2H+QK2 iQ
+QMi`B#mi2 a+2M`BQb- KF2 +?M;2b- M/ }t 2``Q`bX
```
Since every file has its own words, I figured I need to get a piece of translated text, and check if it's present in the target file. One could argue that a simple `grep "Heegu"` would work, but I figured having 3+ words is safer.
So yes, I chose intro for this arbitrarily. Those thee lines are not obfuscated by any LaTeX commands, and the middle one contains no special characters in any language. I figured it's a good pick. I'm open to ideas if there's an obviously better way to do it.

**Comment by qwrtln** [comment]
Yay, it worked for the languages built with LuaTeX. One can assume the test is effective. Now let's try with downgraded Ghostscript.

**Comment by DarthGandalf** [comment]
> I'm open to ideas if there's an obviously better way to do it.

What about letting a language to specify a substring which it guarantees to exist?

e.g. add a line like this to main_en.tex:

```
% TEST_SUBSTRING=Vous êtes tous invités à apporter
```

then grep for that line

**Comment by qwrtln** [comment]
> > I'm open to ideas if there's an obviously better way to do it.
> 
> What about letting a language to specify a substring which it guarantees to exist?
> 
> e.g. add a line like this to main_en.tex:
> 
> ```
> % TEST_SUBSTRING=Vous êtes tous invités à apporter
> ```
> 
> then grep for that line

1. Text from `main_*.tex` is guaranteed to exist.
2. One would need to specify it in the matrix workflow, which GitHub would visualize with every job. I hate this UI. See how it looks in the mission book PR, I'm about to change it.

**Comment by DarthGandalf** [comment]
You misunderstood me.

I said, add that string to main.tex, literally, as a comment. Then in the workflow grep for TEST_STRING (literally), then grep for its result in the pdf

**Comment by DarthGandalf** [comment]
> Text from main_*.tex is guaranteed to exist.

Sure. But "the text exactly 2 lines below 'intro' string" isn't.

**Comment by qwrtln** [comment]
> I said, add that string to main.tex, literally, as a comment. Then in the workflow grep for TEST_STRING (literally), then grep for its result in the pdf

OK, should we repurpose the intro line for this?

**Comment by DarthGandalf** [comment]
No, because you can't guarantee the intro text will have the same arbitrary property in all languages, which happened to be true now.

**Review by tomaas-zeman — APPROVED** [review]

**Pull request merged**